src/inmanta/env.py

```diff
@@ -605,9 +605,10 @@ def check(cls, in_scope: Optional[Pattern[str]] = None, constraints: Optional[Li
         constraint_violations: List[Tuple[Requirement, Optional[version.Version]]] = [
             (constraint, installed_versions.get(constraint.key, None))
             for constraint in all_constraints
-            if constraint.key not in installed_versions or str(installed_versions[constraint.key]) not in constraint
+            if (constraint.key not in installed_versions or str(installed_versions[constraint.key]) not in constraint)
+            and not (constraint.marker and not constraint.marker.evaluate())
```

Comment:
FYI, you can just chain ifs, which can help keep it readable without needing a bunch of parentheses and indentation:
```
l = [
    x
    for x in range(10)
    if x % 2 == 0
    if x > 5
]
assert l == [6, 8]
```